package/system/iucode-tool/patches/001-iucode_tool-add-missing-limits.h-for-USE_CPUID_DEVIC.patch

```diff
@@ -0,0 +1,29 @@
+From e1137cdafc97c0a8b2a0717a771823f3c4320087 Mon Sep 17 00:00:00 2001
+From: Christian Marangi <ansuelsmth@gmail.com>
+Date: Tue, 29 Nov 2022 19:37:03 +0100
+Subject: [PATCH] iucode_tool: add missing limits.h for USE_CPUID_DEVICE
+
+If USE_CPUID_DEVICE is enabled, compilation fails for missing define.
+Add the missing include to fix compilation error with USE_CPUID_DEVICE
+define.
+
+Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
+---
+ iucode_tool.c | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/iucode_tool.c b/iucode_tool.c
+index 4bba5db..0246035 100644
+--- a/iucode_tool.c
++++ b/iucode_tool.c
+@@ -19,6 +19,7 @@
+ #include <stdint.h>
+ #include <sys/types.h>
+ #include <sys/stat.h>
++#include <limits.h>
```

Comment:
This change is not included upstream and there is [no pending pull request ](https://gitlab.com/iucode-tool/iucode-tool/-/merge_requests)to contain it there. This leads to question - why is it not necessary to have it in OpenWrt 22.03 or even in master? Most likely, this is caused by different kernel versions and that's not clear from the commit message.

Reply:
I honestly think the same problem is present upstream and on master but for some reason USE_CPUID_DEVICE is never enabled... I still have to discover why with the 21.02 sdk this config gets applied... 

But the error is there and it's clear... It was never notice for some reason cause the USE_CPUID_DEVICE flag was never tested? I have to do more test with checking if I'm missing something but current idea is that some include define indirectly limits.h but for sdk toolchain this doesn't happen...

Reply:
@BKPepe from what i can see the problem here is not related to kernel version but musl-libc that require explicit declaration of the limits.h define... I think musl fixed this in later version but our 21.02 old version doesn't have this...

Also iucode_microcode.c include limits.h... I will try to propose this change to the original project. Does it make sense to you this reason? I found online multiple similar problem about this same compilation error.

Reply:
Ah the old limits.h include strikes again. I remember a Void Linux patch for some package that fixed compilation with musl by including it. I wondered why.

Then I found out that fortify-headers included it internally and hid the error.

Reply:
oh ok so it's fortify-headers the reason it works... currently old external toolchain (and i assume old sdk) doesn't include fortify headers so this may be the reason I'm encountering this... Think I have to backport the fortify-header patch and see if that actually fix the error... but sdk needs to be rebuild i think?

Reply:
Is it this the related comment?
https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=b21ddbfa18cd1e437e31dc0fa29408721a186c2a

Reply:
Maybe, maybe not.

My comment was based on locally removing fortify-headers and seeing what fails with CONFIG_ALL.

Reply:
@neheb so should we care to fix it anyway (even if the fortify commit works?) 
I think I will send the fixup upstream anyway and see what they say

Reply:
That's what I did in the packages feed for packages missing limits.h.

Reply:
(btw talking ot i have this crazy idea of running everything in a minimal container absolutely min tools installed to check if we are leaking and using some host tools...)

Reply:
If it helps, a recent fresh install of Fedora 37 revealed mosh and libxcrypt in the packages feed rely on host perl-open and perl-diagnostics. Need to patch to use perl/host. There are probably more such problems.

Reply:
@neheb well yes we have a winner... you are totally right. with the backported fortify fix everything compile correctly... 
So i think I will 
- propose the fix upstream
- backport to 21.02 the fortify fix

Reply:
Yes, this seems what needs to be done, thanks @Ansuel!

Reply:
Without b21ddbfa18cd1e437e31dc0fa29408721a186c2a forify sources is not used with external toolchain. We should use it by default. The iucode_tool and some other application (I think strace) are broken in master with musllibc and without fortify sources.

Reply:
I would suggest to fix forify sources in 22.1 and add such a change to OpenWrt master.